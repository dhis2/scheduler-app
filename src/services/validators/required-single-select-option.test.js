import requiredSingleSelectOption from './required-single-select-option'

describe('requiredSingleSelectOption', () => {
    it('should return an error message when there is no input', () => {
        const actual = requiredSingleSelectOption()

        expect(actual).toMatchSnapshot()
    })

    it('should return an error message when input is not an object', () => {
        const actual = requiredSingleSelectOption('not an object')

        expect(actual).toMatchSnapshot()
    })

    it('should return an error message when input has no props', () => {
        const actual = requiredSingleSelectOption({})

        expect(actual).toMatchSnapshot()
    })

    it('should return an error message when input has no value prop', () => {
        const actual = requiredSingleSelectOption({ label: 'label' })

        expect(actual).toMatchSnapshot()
    })

    it('should return an error message when input has no label prop', () => {
        const actual = requiredSingleSelectOption({ value: 'value' })

        expect(actual).toMatchSnapshot()
    })

    it('should return undefined when input is an object with value and label', () => {
        const actual = requiredSingleSelectOption({ value: 1, label: 2 })

        expect(actual).toBeUndefined()
    })
})
