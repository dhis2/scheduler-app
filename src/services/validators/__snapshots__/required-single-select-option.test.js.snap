// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`requiredSingleSelectOption should return an error message when input has no label prop 1`] = `"Please select an option"`;

exports[`requiredSingleSelectOption should return an error message when input has no props 1`] = `"Please select an option"`;

exports[`requiredSingleSelectOption should return an error message when input has no value prop 1`] = `"Please select an option"`;

exports[`requiredSingleSelectOption should return an error message when input is not an object 1`] = `"Please select an option"`;

exports[`requiredSingleSelectOption should return an error message when there is no input 1`] = `"Please select an option"`;
